### PR TITLE
fix runtime error on secret key generation

### PR DESCRIPTION
**Pull request by jpierre03**

### CODE REVIEW

**Comment by domenkozar** [comment]
cc @expipiplus1

File: backend/src/Hercules/ServerEnv.hs [hunk 1]
```diff
@@ -182,14 +182,23 @@ getHerculesConnection Config{..} = liftIO $ do
       pure Nothing
     MigrationSuccess -> pure (Just herculesConnection)
 
--- | Load the key from the secret key file
+-- | Load the key from the secret key file if exists or create one.
```
Comment (expipiplus1) [inline backend/src/Hercules/ServerEnv.hs]: Could this be "if it exists" please

File: backend/src/Hercules/ServerEnv.hs [hunk 2]
```diff
@@ -182,14 +182,23 @@ getHerculesConnection Config{..} = liftIO $ do
...
 getCipher :: MonadIO m => Config -> m (Maybe HerculesCipher)
 getCipher Config{..} = liftIO $ do
+  say ("Trying to open new key at: " <> pack configSecretKeyFile)
```
Comment (expipiplus1) [inline backend/src/Hercules/ServerEnv.hs]: This should probably be `sayErr` as it's a diagnostic message. Eventually we'll hook up monad log into the startup code, but stderr will have to do for now.

File: backend/src/Hercules/ServerEnv.hs [hunk 3]
```diff
@@ -182,14 +182,23 @@ getHerculesConnection Config{..} = liftIO $ do
...
   key <- readFileMaybe configSecretKeyFile >>= \case
     Nothing -> do
       sayErr ("Unable to open secret key file: " <> pack configSecretKeyFile)
+      let herculeCipherBlockSize = blockSize (undefined :: HerculesCipher)
+      let herculeCipherKeySize = case cipherKeySize (undefined :: HerculesCipher) of
+            KeySizeFixed value -> value
+            KeySizeEnum values -> maximum values
+            KeySizeRange _ maxValue -> maxValue
+      sayErr ("Cipher name: " <> pack (show (cipherName (undefined :: HerculesCipher) )) )
```
Comment (expipiplus1) [inline backend/src/Hercules/ServerEnv.hs]: Could you remove the spaces between the brackets at the end of this line, please.

File: backend/src/Hercules/ServerEnv.hs [hunk 4]
```diff
@@ -182,14 +182,23 @@ getHerculesConnection Config{..} = liftIO $ do
...
+      sayErr ("Cipher block size: " <> pack (show herculeCipherBlockSize))
+      sayErr ("Cipher key size: " <> pack (show herculeCipherKeySize))
       sayErr ("Creating new key at: " <> pack configSecretKeyFile)
-      bytes <- getEntropy (blockSize (undefined :: HerculesCipher))
+      bytes <- getEntropy herculeCipherKeySize
       BS.writeFile configSecretKeyFile bytes
       pure bytes
```
Comment (expipiplus1) [inline backend/src/Hercules/ServerEnv.hs]: I think that the key generation/writing code is long enough now to warrant its own function. Could you move the code on lines 191-203 to a function called something like `generateNewKey :: MonadIO m => Config -> m (Key HerculesCipher)`.

Thanks

**Comment by jpierre03** [comment]
@expipiplus1 PR updated

**Comment by expipiplus1** [comment]
Thanks!

**Pull request merged**